pkg/pebinaryexecutor/pebinary.go

```diff
@@ -0,0 +1,268 @@
+		return err
+	}
+
+	if p.IsServer {
+		return bootstrap.UpdateManifests(p.Resolver, nodeConfig, cfg)
+	}
+
+	restConfig, err := clientcmd.BuildConfigFromFlags("", nodeConfig.AgentConfig.KubeConfigK3sController)
+	cniType, err := getCniType(restConfig)
+	if err != nil {
+		return err
+	}
+
+	switch cniType {
+	case "calico":
+		p.cni = &win.Calico{}
+	default:
+		return fmt.Errorf("the CNI %s isn't supported on Windows", cniType)
+	}
+
+	cniConfig, err := p.cni.Setup(ctx, p.DataDir, nodeConfig, restConfig)
+	if err != nil {
+		return err
+	}
+	p.cniConig = cniConfig
+
+	logrus.Infof("Okay, exiting setup.")
+	return nil
+}
+
+// Kubelet starts the kubelet in a subprocess with watching goroutine.
+func (p *PEBinary) Kubelet(args []string) error {
+	extraArgs := map[string]string{
+		"file-check-frequency":     "5s",
+		"sync-frequency":           "30s",
+		"cgroups-per-qos":          "false",
+		"enforce-node-allocatable": "",
+		"resolv-conf":              "",
+		"hairpin-mode":             "promiscuous-bridge",
+	}
+	if p.CloudProvider != nil {
+		extraArgs["cloud-provider"] = p.CloudProvider.Name
+		extraArgs["cloud-config"] = p.CloudProvider.Path
+	}
+
+	if err := hcn.DSRSupported(); err == nil {
+		extraArgs["feature-gates"] = addFeatureGate(extraArgs["feature-gates"], "WinDSR=true")
+	}
+
+	args = append(getArgs(extraArgs), args...)
+	logrus.Infof("Running RKE2 kubelet %s", args)
+	go func() {
+		for {
+			cmd := exec.Command(p.KubeletPath, args...)
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+
+			err := cmd.Run()
+			logrus.Errorf("Kubelet exited: %v", err)
+
+			time.Sleep(5 * time.Second)
+		}
+	}()
+
+	return p.cni.Start(p.cniConig)
+}
+
+// KubeProxy starts the kubeproxy in a subprocess with watching goroutine.
+func (p *PEBinary) KubeProxy(args []string) error {
+	extraArgs := map[string]string{
+		"hostname-override": p.cniConig.CalicoConfig.Hostname,
+		"v":                 "4",
+		"proxy-mode":        "kernelspace",
+		"kubeconfig":        p.cniConig.NodeConfig.AgentConfig.KubeConfigKubeProxy,
+		"network-name":      p.cniConig.NetworkName,
+		"bind-address":      p.cniConig.BindAddress,
+	}
+
+	if err := hcn.DSRSupported(); err == nil {
+		logrus.Infof("WinDSR support is enabled")
+		extraArgs["feature-gates"] = addFeatureGate(extraArgs["feature-gates"], "WinDSR=true")
+		extraArgs["enable-dsr"] = "true"
+	}
+
+	for range time.Tick(time.Second * 5) {
+		endpoint, err := hcsshim.GetHNSEndpointByName("Calico_ep")
+		if err != nil {
+			logrus.WithError(err).Warning("can't find %s, retrying", "Calico_ep")
+			continue
+		}
+		extraArgs["source-vip"] = endpoint.IPAddress.String()
+		break
+	}
+
+	logrus.Infof("Deleting HNS policies before kube proxy starts.")
+	policies, _ := hcsshim.HNSListPolicyListRequest()
+	for _, policy := range policies {
+		policy.Delete()
+	}
+
+	args = append(getArgs(extraArgs), args...)
+	logrus.Infof("Running RKE2 Kubeproxy %s", args)
```

Comment:
You're going to get multiple log lines for this, since the k3s code also logs the args before passing them into the executor... but we should at least be consistent about what we call it.
```suggestion
	logrus.Infof("Running kube-proxy %s", args)
```